### PR TITLE
Update docs to reflect new automated process for feature grouping

### PR DESCRIPTION
**Pull request by getdave**
## Description
In https://github.com/WordPress/gutenberg/pull/33229 we added automation to the release tooling to attempt to auto-categorize the grouping of PRs by "feature".

This PR updates the docs to reflect that and to add information on exactly how you should look to curate the changelog in response to what the script will output.

## How has this been tested?
Reading and checking for accuracy.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by talldan — APPROVED** [review]
Looks good. The only other thing I did was to ping a few folks who'd worked on, reviewed or merged an uncategorized PR to let them know why the labels are important.

It'd be good to have an automated task do this, but in the meantime it could also be worth mentioning in the docs.

**Comment by gziolo** [comment]
> The only other thing I did was to ping a few folks who'd worked on, reviewed or merged an uncategorized PR to let them know why the labels are important.

Could we add a logic to `Pull request automation` workflow that fails the check if there are no labels? Ideally, one of them should be prefixed with `[Type]` and the other one should map to one of the feature groups.

File: docs/contributors/code/release.md [hunk 1]
```diff
@@ -62,17 +73,23 @@ Documenting the release is a group effort between the release manager, Gutenberg
...
 
+-   Move _all_ entries under the `Various` section to a more appropriate section.
+-   Move _all_ features listed under the `Uncategorized` bullet points to a more suitable feature grouping.
 -   Fix spelling errors or clarify wording. Phrasing should be easy to understand where the intended audience is those who use the plugin or are keeping up with ongoing development.
 -   Create new groupings as applicable, and move pull requests between.
 -   When multiple pull requests relate to the same task (such as a follow-up pull request), try to combine them to a single entry.
 -   If subtasks of a related set of pull requests are substantial, consider organizing as entries in a nested list.
 -   If one or more pull requests revert one or more pull requests in the same release netting a zero-sum of code changes, remove them.
--   Remove mobile app pull request entries.
+-   Remove all mobile app pull request entries.
```
Comment (gziolo) [inline docs/contributors/code/release.md]:
```suggestion
-   Remove all mobile app pull request entries.
```

There is one exception, if the mobile app related PR updates also funcionality for the web. This is how we missed React upgrade from v16 to v17 in the plugin changelog 😅

**Review by gziolo — APPROVED** [review]

File: docs/contributors/code/release.md (hunk 1 above)
Comment (getdave) [reply docs/contributors/code/release.md]: I've updated the docs to reflect this 👍

**Pull request merged**